components/property_access/introduction.rst

```diff
@@ -365,8 +371,226 @@ You can also mix objects and arrays::
     var_dump('Hello '.$accessor->getValue($person, 'children[0].firstName')); // 'Wouter'
     // equal to $person->getChildren()[0]->firstName
 
+Custom method calls and virtual properties in a class
+-----------------------------------------------------
+
+Sometimes you may not want the component to guess which method has to be called
+when reading or writing properties. This is specially interesting when property
+names are not in English or its singularization is not properly detected.
+
+For those cases you can add metadata to the class being accessed so that the
+component will use a particular method as a getter, setter or even adder and
+remover (for collections).
+
+Another interesting use of custom methods is declaring virtual properties
+which are not stored directly in the object.
+
+There are three supported ways to state this metadata supported out-of-the-box by
+the component: using annotations, using YAML configuration files or using XML
+configuration files.
+
+.. caution::
+
+    When using as a standalone component the metadata feature is disabled by
+    default. You can enable it by calling
+    :method:`PropertyAccessorBuilder::setMetadataFactory
+    <Symfony\\Component\\PropertyAccess\\PropertyAccessorBuilder::setMetadataFactory>`
+    see `Enable other Features`_.
+
+There are four method calls that can be overriden: `getter`, `setter`, `adder` and
+`remover`.
+
+When using annotations you can precede a property with `@Property` to state which
+method should be called when a get, set, add or remove operation is needed on the
+property.
+
+.. configuration-block::
+
+    .. code-block:: php
+
+        // ...
+        use Symfony\Component\PropertyAccess\Annotation\Property;
+
+        class Person
+        {
+            /**
+             * @Property(getter="getFullName", setter="setFullName")
+             */
+            private $name;
+
+            /**
+             * @Property(adder="addNewChild", remover="discardChild")
+             */
+            private $children;
+
+            public function getFullName()
+            {
+                return $this->name;
+            }
+
+            public function setFullName($fullName)
+            {
+                $this->name = $fullName;
+            }
+        }
+
+    .. code-block:: yaml
+
+        Person:
```

Comment:
At the beginning of this file, we need the path of the example file (using AppBundle if possible). Example:

```yaml
# src/AppBundle/Resources/config/doctrine/Person.orm.yml
Person:
    name:
# ...
```